System/System.cpp

```diff
@@ -20,9 +22,22 @@ namespace RTE {
 	const std::string System::s_ZippedModulePackageExtension = ".rte.zip";
 	const std::unordered_set<std::string> System::s_SupportedExtensions = { ".ini", ".txt", ".lua", ".cfg", ".bmp", ".png", ".jpg", ".jpeg", ".wav", ".ogg", ".mp3", ".flac" };
 
-/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+#ifdef __unix__
+	const std::filesystem::path System::s_BaseDataDirectory{BASEDATAPATH};
+	std::filesystem::path System::s_TempDirectory{};
+#endif
+
+	/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
 	void System::Initialize() {
+#if defined(__unix__) && !LINUX_PORTABLE
+		char dirTemplate[]{"/tmp/CCCP.XXXXXX"};
+		s_TempDirectory = std::filesystem::path(mkdtemp(dirTemplate));
+		SetupBaseGameFolders(s_TempDirectory);
+		SetupUserFolders(s_TempDirectory);
+		std::filesystem::current_path(s_TempDirectory);
```

Comment:
What exactly is the point of `s_TempDirectory` if you're setting it as the working dir here?
Can't you just
```cpp
	std::filesystem::path tempDir = std::filesystem::path(mkdtemp("/tmp/CCCP.XXXXXX"));
	SetupBaseGameFolders(tempDir);
	SetupUserFolders(tempDir);
	std::filesystem::current_path(tempDir);
```
or without `mkdtemp`
```cpp
	std::filesystem::path tempDir = std::filesystem::temp_directory_path() / ("CCCP." + std::to_string(RandomNum(000000, 999999)));
	MakeDirectory(tempDir.generic_string());
	SetupBaseGameFolders(tempDir);
	SetupUserFolders(tempDir);
	std::filesystem::current_path(tempDir);
```
and then when you destroy just `std::filesystem::remove_all(s_WorkingDirectory);`?